### DISCUSSION TITLE
Add a "classic mode" / Performance Mode

### DISCUSSION THREAD

**Discussion by godkama** [ppy/osu#29725] [Ideas]
These are two different ideas. I'll clarify

### "classic mode"

Add a setting in the options menu called *"Classic UI"* with a on/off toggle.

This will work as follows :
- You will have the same song select menu as classic 
- All animations will be from classic
- The play screen will look the same as osu!classic

This is an idea for people still reticent to switch to lazer due to overcomplication. They will be able to enjoy their famously known look and enjoy lazer to it's fullest !

### "Performance Mode"

Add a setting to the options menu called "Performance Mode" which will be powered by an on/off switch 

That works like this :
- Optimize memory usage
- Force high priority
- Disable most extra animations
- Revert to classic animations for smoother gameplay
- Use less images

This is useful for people (like me) who get average 200 fps on classic but end up with only 30 fps on Lazer making it unplayable. I recon something like this will bring quite a few extra players.

**Comment by Tenexxt** [comment]
Classic Mode would probably be possible if the game UI would be highly skinnable (moving parts around, changing icons and some other objects like the osu! cookie, etc.), which I hope will be possible in the future (even if it takes years).

As for performance mode, this sounds like a good feature for weaker devices, tho I don't know how switching to osu!stable animations would boost performance. Would make more sense to just disable them completely since it's a **performance** mode.

**Reply by godkama** [reply]
Well i don't know how this all works in detail but stable animations were built on weaker engines in the beginning and are thus somewhat smaller / less hq, therefore consuming less. I also think theres a possibility to make a second client that will launch on perf mode and that will have no jibber jabber, no ingame browser and extra plugins etc.

**Reply by godkama** [reply]
Or they could just make a thing similar to -devserver allowing you to connect to lazer from stable
